src/update_xs_yum.ml

```diff
@@ -470,27 +469,26 @@ let get_last_successful_build branch =
     (get_env_var "JENKINS_URL") // path
   in
   get_http_body url
-  >>|= fun body ->
+  >>= fun body ->
   let open Yojson.Basic in
   Lwt.catch
     (fun () ->
-      `Ok (from_string body
-          |> Util.member "lastSuccessfulBuild"
-          |> Util.member "number"
-          |> to_string)
-      |> Lwt.return)
+       (from_string body
+        |> Util.member "lastSuccessfulBuild"
+        |> Util.member "number"
+        |> to_string)
+       |> Lwt.return)
     (fun exn ->
-      let msg = Printf.sprintf "Error: Getting latest build number failed for branch %s: %s\n%!" branch (Printexc.to_string exn) in
-      Lwt_io.eprintl msg >>= fun () ->
-      `Error msg |> Lwt.return)
+       let msg = Printf.sprintf "Error: Getting latest build number failed for branch %s: %s\n%!" branch (Printexc.to_string exn) in
+       fail_with_log msg)
 
 let print_whitelist () =
   Printf.printf "Using whitelist:\n%!";
   List.iter (fun x -> Printf.printf "%s\n%!" x) whitelist
 
 let best_effort_upload branch f =
   Lwt.catch
-    (fun () -> f () >>|= fun () -> Lwt.return_unit)
+    f
```

Comment:
would it be nicer to have `Lwt.catch f` in the same line or even `Lwt.catch f (fun ex ->`?

Reply:
Makes sense, I can do that if it makes it more readable.

Reply:
Done.